app-core-test/src/test/java/net/spals/appbuilder/app/core/generic/SampleGenericWorkerAppFTest.java

```diff
@@ -111,15 +121,28 @@ public void testFileStoreInjection() {
         assertThat(fileStorePluginMap, hasKey("localFS"));
     }
 
+    @Test
+    public void testKeyStoreInjection() {
+        final Injector serviceInjector = sampleApp.getServiceInjector();
+        assertThat(serviceInjector.getInstance(KeyStore.class), notNullValue());
+
+        final TypeLiteral<Map<String, KeyStorePlugin>> keyStorePluginMapKey =
+            new TypeLiteral<Map<String, KeyStorePlugin>>(){};
+        final Map<String, KeyStorePlugin> keyStorePluginMap =
+            serviceInjector.getInstance(Key.get(keyStorePluginMapKey));
+        assertThat(keyStorePluginMap, aMapWithSize(1));
+        assertThat(keyStorePluginMap, hasKey("password"));
+    }
+
     @Test
     public void testMapStoreInjection() {
         final Injector serviceInjector = sampleApp.getServiceInjector();
         assertThat(serviceInjector.getInstance(MapStore.class), notNullValue());
 
         final TypeLiteral<Map<String, MapStorePlugin>> mapStorePluginMapKey =
-                new TypeLiteral<Map<String, MapStorePlugin>>(){};
+            new TypeLiteral<Map<String, MapStorePlugin>>(){};
```

Comment:
nit pick, can this be new TypeLiteral<>() {}... oh its an anonymous class. So if you aren't aware, my understanding is that java9 will add better type inference so you won't have to clarify anonymous classes.

Reply:
But as this is still java8 we need to, just thought it was cool to mention and keep hoping for getting on java9 soon.